brainstorming.md

```diff
@@ -0,0 +1,372 @@
+
+Implementations _should_ enforce the naming and indexing constraints on these pre-defined fields. Pre-defined fields are distinguished from other user-defined collection elements, where no schema or indexing semantics are presumed or enforced.
+
+The predefined fields within SOMA share common indices and `shape`:
+
+- obs_id - the observation, a `string` type
+- var_id - the variable, a `string` type
+
+SOMA pre-defined fields are principally differentiated by:
+
+- dimensionality, eg, 1-D (SOMADataFrame) or 2-D (SOMAMatrix)
+- index column names, `obs_id` or `var_id` or both
+
+The pre-defined fields are:
+
+| Field name | Field type                              | Field description                                                                                                                                                                                                                                               |
+| ---------- | --------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `obs`      | `SOMADataFrame`                         | Primary annotations on the _observation_ axis. Must be of length #observations (ie, every observation _must_ be defined in this field), and must contain and be indexed by the `obs_id` column. The contents of `obs_id` define the _observation_ index domain. |
+| `var`      | `SOMADataFrame`                         | Primary annotations on the _variable_ axis. Must be of length #variables (ie, every variable _must_ be defined in this field), and must contain and be indexed by the `var_id` column. The contents of `var_id` define _variable_ index domain.                 |
+| `obsm`     | `SOMACollection[string, SOMADataFrame]` | Secondary _observation_ annotations, indexed with `obs_id`                                                                                                                                                                                                      |
+| `varm`     | `SOMACollection[string, SOMADataFrame]` | Secondary _variable_ annotations, indexed with `var_id`                                                                                                                                                                                                         |
+| `obsp`     | `SOMACollection[string, SOMAMatrix]`    | Pairwise _observation_ annotations, indexed with `[obs_1_id, obs_2_id]`                                                                                                                                                                                         |
```

Comment:
ok here is fine -- above i said maybe we should say these explicitly -- you're doing it here -- this is fine